### PR TITLE
Fix for crash when SGReplicator does not waits for C4Replicator

### PR DESCRIPTION
**Pull request by RastislavKamenickyOnsemi**

### CODE REVIEW

**Comment by victorbergeronsemi** [comment]
Tested, looks good to me.

**Review by victorbergeronsemi — APPROVED** [review]

File: src/SGReplicator.cpp [hunk 1]
```diff
@@ -68,13 +68,31 @@ namespace Strata {
...
+            if(c4replicator_ != nullptr) {
+                while (c4repl_getStatus(c4replicator_).level != kC4Stopped)
+                    this_thread::sleep_for(chrono::milliseconds(1));
+            }
+        }
+
+        // then wait for the onStatusChanged event to be emitted and terminated
+        unsigned count = 0;     // in case the C4Replicator would fail to emit onStatusChanged event
+        while ((internal_status_ != Strata::SGReplicatorInternalStatus::kStopped) && (count < 200)) {
+            this_thread::sleep_for(chrono::milliseconds(1));
```
Comment (victorbergeronsemi) [inline src/SGReplicator.cpp:91]: Sleeping for only 1 ms might be too short - consider sleeping for longer and reducing the max count

Comment (RastislavKamenickyOnsemi) [reply src/SGReplicator.cpp:91]: I checked and it usually takes around ~1 msec to complete the operations under normal scenarios.
Should I change sleep to 5 msec and max count to for example 40?

Comment (victorbergeronsemi) [reply src/SGReplicator.cpp:91]: In this case since you checked/timed the operation, I believe this might be fine then. I'd say it's up to you.

**Review by luayalshawionsemi — APPROVED** [review]

**Pull request merged**